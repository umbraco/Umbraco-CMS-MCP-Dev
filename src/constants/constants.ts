export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
