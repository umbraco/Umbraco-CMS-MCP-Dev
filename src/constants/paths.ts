import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');