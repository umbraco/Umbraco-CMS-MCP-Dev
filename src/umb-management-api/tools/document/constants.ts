export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}