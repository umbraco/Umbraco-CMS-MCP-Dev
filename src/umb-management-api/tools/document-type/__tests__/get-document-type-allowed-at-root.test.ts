import GetDocumentTypeAllowedAtRootTool from "../get/get-document-type-allowed-at-root.js";
import { DocumentTypeBuilder } from "./helpers/document-type-builder.js";
import { DocumentTypeTestHelper } from "./helpers/document-type-test-helper.js";
import { DocumentTypeResponseModel } from "@/umb-management-api/schemas/index.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";

const TEST_DOCTYPE_NAME = "_Test DocumentType Root";

describe("get-document-type-allowed-at-root", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    // Clean up any test document types
    await DocumentTypeTestHelper.cleanup(TEST_DOCTYPE_NAME);
  });

  it("should get document types allowed at root", async () => {
    // Create test document types
    await new DocumentTypeBuilder()
      .withName(TEST_DOCTYPE_NAME)
      .withIcon("icon-document")
      .allowAsRoot(true)
      .create();

    // Get document types allowed at root
    const result = await GetDocumentTypeAllowedAtRootTool().handler(
      {
        skip: 0,
        take: 10,
      },
      { signal: new AbortController().signal }
    );

    // Parse and find our test document type
    const parsed = JSON.parse(result.content[0].text as string) as {
      items: DocumentTypeResponseModel[];
    };
    const testDocType = parsed.items.find(
      (item) => item.name === TEST_DOCTYPE_NAME
    );

    if (!testDocType) {
      throw new Error("Test document type not found in results");
    }

    // Normalize the ID
    testDocType.id = BLANK_UUID;

    // Verify just the test document type
    expect(testDocType).toMatchSnapshot();
  });
});
