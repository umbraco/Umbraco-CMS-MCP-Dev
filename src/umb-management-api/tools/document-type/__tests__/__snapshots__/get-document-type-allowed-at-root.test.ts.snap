// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-document-type-allowed-at-root should get document types allowed at root 1`] = `
{
  "description": null,
  "icon": "icon-document",
  "id": "00000000-0000-0000-0000-000000000000",
  "name": "_Test DocumentType Root",
}
`;
